File: solution/35/solution.py
class Solution(object):
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        index = -1
        begin = 0
        end = len(nums) -1
        while index == -1:
            mid = (begin+end)/2
            if nums[mid] == target:index = mid
            if nums[mid] > target:
                if mid == 0 or nums[mid - 1] < target:
                    index = mid
                end = mid -1
            if nums[mid] < target:
                if mid == len(nums) -1 or nums[mid + 1] > target:
                    index = mid + 1
                begin = mid + 1
        return index

File: solution/35/solution.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        return midfind(0, nums.size() - 1, target, nums);
    }

    int midfind(int left, int right, int target, vector<int>& nums){
        int mid = (left + right) / 2;
        if(nums[mid] == target)return mid;
        if(nums[mid] > target){
            if(mid==0||target<nums[mid-1])return mid;
            return midfind(left, mid-1, target, nums);
        }
        else{
            if(mid==nums.size()-1||target<nums[mid+1])return mid +1;
            return midfind(mid+1, right, target, nums);
        }
    }
};

